pyinfra/operations/docker.py

```diff
@@ -0,0 +1,339 @@
+            volumes=["nginx_data:/usr/share/nginx/html"],
+            pull_always=True,
+        )
+
+        # Stop a container
+        docker.container(
+            name="Stop Nginx container",
+            container="nginx",
+            start=False,
+        )
+
+        # Start a container
+        docker.container(
+            name="Start Nginx container",
+            container="nginx",
+            start=True,
+        )
+    """
+
+    existent_container = [c for c in host.get_fact(DockerContainers) if container in c["Name"]]
+
+    if force:
+        if existent_container:
+            yield handle_docker(
+                resource="container",
+                command="remove",
+                container=container,
+            )
+
+    if present:
+        if not existent_container or force:
+            yield handle_docker(
+                resource="container",
+                command="create",
+                container=container,
+                image=image,
+                ports=ports,
+                networks=networks,
+                volumes=volumes,
+                env_vars=env_vars,
+                pull_always=pull_always,
+                present=present,
+                force=force,
+                start=start,
+            )
+
+    if existent_container and start:
+        if existent_container[0]["State"]["Status"] != "running":
+            yield handle_docker(
+                resource="container",
+                command="start",
+                container=container,
+            )
+
+    if existent_container and not start:
+        if existent_container[0]["State"]["Status"] == "running":
+            yield handle_docker(
+                resource="container",
+                command="stop",
+                container=container,
+            )
+
+    if existent_container and not present:
+        yield handle_docker(
+            resource="container",
+            command="remove",
+            container=container,
+        )
+
+
+@operation(is_idempotent=False)
+def image(image, present=True):
+    """
+    Manage Docker images
+
+    + image: Image and tag ex: nginx:alpine
+    + present: whether the Docker image should be exist
+
+    **Examples:**
+
+    .. code:: python
+
+        # Pull a Docker image
+        docker.image(
+            name="Pull nginx image",
+            image="nginx:alpine",
+            present=True,
+        )
+
+        # Remove a Docker image
+        docker.image(
+            name="Remove nginx image",
+            image:"nginx:image",
+            present=False,
+        )
+    """
+
+    if present:
+        yield handle_docker(
+            resource="image",
+            command="pull",
+            image=image,
+        )
+
+    else:
+        yield handle_docker(
+            resource="image",
+            command="remove",
+            image=image,
+        )
+
+
+@operation()
+def volume(volume, driver="", labels=[], present=True):
```

Comment:
Default list here

Reply:
resolved in commit [0ea1976](https://github.com/pyinfra-dev/pyinfra/pull/1114/commits/0ea1976f4fa2881d636e952a0f2018e891d47434)